bittide-instances/src/Bittide/Instances/Tests/FullMeshHwCc/IlaPlot.hs

```diff
@@ -0,0 +1,268 @@
+-- SPDX-FileCopyrightText: 2023 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+{-# LANGUAGE MultiWayIf #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE FlexibleContexts #-}
+
+{-# OPTIONS_GHC -fconstraint-solver-iterations=20 #-}
+module Bittide.Instances.Tests.FullMeshHwCc.IlaPlot
+  ( PlotData(..)
+  , IlaControl(..)
+  , RfStageChange(..)
+  , CaptureCondition(..)
+  , GlobalTimestamp
+  , LocalTimestamp
+  , AccWindowHeight
+  , SyncPulsePeriod
+  , syncOutGenerator
+  , callistoClockControlWithIla
+  ) where
+
+import Clash.Explicit.Prelude
+import Clash.Explicit.Signal.Extra
+
+import Bittide.Arithmetic.Time (Milliseconds, PeriodToCycles)
+import Bittide.ClockControl (SpeedChange(..), DataCount, ClockControlConfig)
+import Bittide.ClockControl.Callisto
+  (CallistoResult(..), ReframingState(..), callistoClockControl)
+import Bittide.ClockControl.StabilityChecker
+import Bittide.Extra.Maybe (orNothing)
+
+import Clash.Cores.Xilinx.Xpm.Cdc.Handshake.Extra (xpmCdcMaybeLossy)
+import Clash.Cores.Xilinx.Xpm.Cdc.Gray (xpmCdcGray)
+import Clash.Cores.Xilinx.Ila (IlaConfig(..), Depth(..), ila, ilaConfig)
+
+import Data.Maybe (isJust, fromMaybe)
+
+type AccWindowHeight = 3 :: Nat
+type SyncPulsePeriod = Milliseconds 5
+type SyncPulseIndex dom = Index (PeriodToCycles dom SyncPulsePeriod)
+
+-- | A global timestamp consists of the number of synchronized pulses
+-- received and the number of cycles of the local stable clock
```

Comment:
Do you reset this value every time you get a global tick? It seems to me that a good timestamp would be `(global ticks, local ticks since last global tick)`. What does continued counting get you?

Reply:
Don't get confused, this is the timer for producing the synchronized plot.